### ISSUE TITLE
Agent uuid not accepted in yaml Environment settings

### ISSUE THREAD

**Issue by jaheikki** [tomzo/gocd-yaml-config-plugin#42]
Hi, I would like to use Environment setting in gocd-yaml, having agent uuid (GUID) defined. The Readme of gocd-yaml-config-plugin tells that user cannot find GUID? I found GUID file from agent disk /godata/config/guid.txt, having this kind of content "f146f42d-b741-4fc2-9dcd-36abb407xxyy". 

However the GUID I found is not accepted by Go: "Environment 'installations' has an invalid agent uuid 'f146f42d-b741-4fc2-9dcd-36abb407...". Also it is not possible to set agent in GoCD UI in Environment section, GoCD says "There are no agents available".

Currently I don't know what is the root problem here, but would it be possible to get this issue fixed somehow?

Clip from the Readme.md:
"Environment

NOTE: The agents should be a guid, which is currently impossible to get for user

testing:
  environment_variables:
    DEPLOYMENT: testing
  secure_variables:
    ENV_PASSWORD: "s&Du#@$xsSa"
  pipelines:
    - example-deploy-testing
    - build-testing
  agents:
    - 123"

**Comment by jaheikki** [comment]
Actually I was able to configure the static agent of mine in Environment settings in GoCD UI after I had enabled the agent in UI manually. However the Environment functionality does not work as I expected in UI either, since the elastic agents appeared automatically to Environments agents. I was expecting that pipelines configured to Environments would have used only one agent (= the static agent). Anyway it seems that  this issue is still valid since agent setting under Environment in yml is not working correctly ('invalid agent uuid' error unless agent is enabled manually in GoCD UI).

**Comment by jaheikki** [comment]
Update: I'm using Go Version: 17.10.0. The problem is not anymore that GUID is not accepted, it is accepted in case my static go agent is enabled in GoCD UI. However, 2 problems still:
1. The static agent I've configured in Environments section in yml will not ever appear in Environments menu in GoCD UI (=agent is not used for Environment pipelines, see figure)?:
![image](https://user-images.githubusercontent.com/8567508/30845102-7f911130-a29a-11e7-9e7d-ad61b444da5a.png)
2. I have go elastic agents in use and they will be automatically set as Environment agents when looked in GoCD UI, which is strange (see figure where none of the agents visible are defined in yml). Notice that currently I can set my static agent manually as Environment agent in GoCD UI, but it will not be used because the jobs will be run in automatically set elastic agents. Maybe this is GoCD bug and not yaml plugin bug?
![image](https://user-images.githubusercontent.com/8567508/30845440-1a249342-a29c-11e7-8273-8a8ea56618b8.png)

My yml file content:
![image](https://user-images.githubusercontent.com/8567508/30845624-f724e21a-a29c-11e7-88cf-b99776f251fd.png)

**Comment by jaheikki** [comment]
Do not know whether these problems related to this issue are GoCD problems rather than yaml plugin problems, so I created issue in GoCD github also: https://github.com/gocd/gocd/issues/3890

**Comment by jaheikki** [comment]
Now that we have investigated more of this, it seems almost everything is working if configurations are created correctly :) My problem was  that I had configured elastic agent in my job causing elastic agent to be used in my Environment instead of static agent. In other words the static agent configured in Environment yml (see figure above) is used correctly if job contains no elastic agent. 

But 2 minor problem still exists:
1.  The static agent id is not shown in GoCD UI in Enviroments tab, even the agent is  :
![image](https://user-images.githubusercontent.com/8567508/31070701-04834ab0-a76a-11e7-8a2d-aef23d7fbed4.png)
2. Environment section documentation (https://github.com/tomzo/gocd-yaml-config-plugin) is unclear/not valid: "NOTE: The agents should be a guid, which is currently impossible to get for user"?
